fcrepo-jms/src/main/java/org/fcrepo/jms/headers/DefaultMessageFactory.java

```diff
@@ -79,36 +93,23 @@ private void setBaseURL(final FedoraEvent event) {
                     url = url.substring(0, url.length() - 1);
                 }
                 this.baseURL = url;
-                LOGGER.debug("MessageFactory baseURL: {}", baseURL);
+                this.userAgent = json.get("userAgent").getAsString();
+                LOGGER.debug("MessageFactory baseURL: {}, userAgent: {}", baseURL, userAgent);
 
             } else {
-                LOGGER.warn("MessageFactory baseURL is empty!");
+                LOGGER.warn("MessageFactory event UserData is empty!");
             }
 
         } catch ( final Exception ex ) {
```

Comment:
JsonParser.parse throws JsonParseException, but the accessors throw IllegalStateExceptions -- so we could narrow to RuntimeException.